shared/middleware/files_api.rb

```diff
@@ -592,6 +592,10 @@ def copy_file(endpoint, encrypted_channel_id, filename, source_filename)
   # manifest.
   #
   def files_put_file(encrypted_channel_id, filename, body)
+    # Rename .jfif file extensions to .jpg because Sinatra does not support .jfif file types.
+    # .jfif files use the same protocol as .jpg files, which is why rename-only is sufficient.
+    filename.sub!(/(.jfif|.JFIF)/, '.jpg') if File.extname(filename.downcase) == '.jfif'
```

Comment:
Is it possible/valid for a filename to have `.jfif` somewhere in the filename other than the extension? if yes, would subbing `.jpg` cause problems?

Reply:
i don't think that's a valid filename, but it would cause problems if this did somehow happen:

```ruby
[development] dashboard > s = "image.jfifhello.jfif"
=> "image.jfifhello.jfif"
[development] dashboard > File.extname(s)
=> ".jfif"
[development] dashboard > s.sub(".jfif", ".jpg")
=> "image.jpghello.jfif"
```

so i'll refactor this to be safe

Reply:
the main issue with the code above is that we aren't globally replacing ".jfif" with ".jpg" so we end up with this weird mishmash of file extensions (i should've used `.gsub!` instead of `.sub!`)